pkg/virt-operator/resource/apply/apps.go

```diff
@@ -50,32 +57,34 @@ func (r *Reconciler) syncDeployment(deployment *appsv1.Deployment) error {
 	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, deployment.ObjectMeta)
 
 	// there was no change to metadata, the generation matched
-	if !*modified && existingCopy.GetGeneration() == expectedGeneration {
+	if !*modified &&
+		*deployment.Spec.Replicas == *origDeployment.Spec.Replicas &&
+		existingCopy.GetGeneration() == expectedGeneration {
 		log.Log.V(4).Infof("deployment %v is up-to-date", deployment.GetName())
-		return nil
+		return origDeployment, nil
```

Comment:
@jean-edouard how would you feel about returning the copied `deployment` instead of the origDeployment. I'm just paranoid about us inadvertently corrupting the informer cache at some point in the future. I'd sleep better if this function consistently returned a copy